From: "Mike Rapoport (IBM)" <rppt@kernel.org>
Subject: mm: move free_area_empty() to mm/internal.h
Date: Sun, 26 Mar 2023 19:02:15 +0300

The free_area_empty() helper is only used inside mm/ so move it there to
reduce noise in include/linux/mmzone.h

Link: https://lkml.kernel.org/r/20230326160215.2674531-1-rppt@kernel.org
Signed-off-by: Mike Rapoport (IBM) <rppt@kernel.org>
Suggested-by: Matthew Wilcox (Oracle) <willy@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/mmzone.h |    5 -----
 mm/internal.h          |    5 +++++
 2 files changed, 5 insertions(+), 5 deletions(-)

--- a/include/linux/mmzone.h~mm-move-free_area_empty-to-mm-internalh
+++ a/include/linux/mmzone.h
@@ -110,11 +110,6 @@ struct free_area {
 	unsigned long		nr_free;
 };
 
-static inline bool free_area_empty(struct free_area *area, int migratetype)
-{
-	return list_empty(&area->free_list[migratetype]);
-}
-
 struct pglist_data;
 
 #ifdef CONFIG_NUMA
--- a/mm/internal.h~mm-move-free_area_empty-to-mm-internalh
+++ a/mm/internal.h
@@ -517,6 +517,11 @@ void init_cma_reserved_pageblock(struct
 int find_suitable_fallback(struct free_area *area, unsigned int order,
 			int migratetype, bool only_stealable, bool *can_steal);
 
+static inline bool free_area_empty(struct free_area *area, int migratetype)
+{
+	return list_empty(&area->free_list[migratetype]);
+}
+
 /*
  * These three helpers classifies VMAs for virtual memory accounting.
  */
_
